lldb-eval/eval_test.cc

```diff
@@ -492,6 +492,40 @@ TEST_F(EvalTest, TestPointerArithmetic) {
                       "'const char *')"));
 }
 
+TEST_F(EvalTest, PointerIntegerComparison) {
+  EXPECT_THAT(Eval("(void*)0 == 0"), IsEqual("true"));
+  EXPECT_THAT(Eval("0 != (void*)0"), IsEqual("false"));
+
+  EXPECT_THAT(Eval("(void*)0 == nullptr"), IsEqual("true"));
+  EXPECT_THAT(Eval("(void*)0 != nullptr"), IsEqual("false"));
+  EXPECT_THAT(Eval("nullptr == (void*)1"), IsEqual("false"));
+  EXPECT_THAT(Eval("nullptr != (void*)1"), IsEqual("true"));
+
+  EXPECT_THAT(Eval("nullptr == nullptr"), IsEqual("true"));
+  EXPECT_THAT(Eval("nullptr != nullptr"), IsEqual("false"));
+  EXPECT_THAT(Eval("nullptr == 0"), IsEqual("true"));
+  EXPECT_THAT(Eval("0 != nullptr"), IsEqual("false"));
+
+  EXPECT_THAT(
+      Eval("(void*)0 > nullptr"),
+      IsError(
+          "invalid operands to binary expression ('void *' and 'nullptr_t')"));
+
+  // These are not allowed by C++, but we support it as an extension.
+  EXPECT_THAT(Eval("(void*)1 == 1"), IsEqual("true"));
```

Comment:
Do we have tests for comparing non-zero void* to another non-zero void*?

Reply:
There are a few, but I've added more.